analysis/study_definition.py

```diff
@@ -1,13 +1,884 @@
+  ),
+  # # sotrovimab_approved=covid_therapeutics_approved("Sotrovimab"),
+  # # molnupiravir_approved=covid_therapeutics_approved("Molnupiravir"),
+  # # paxlovid_approved=covid_therapeutics_approved("Paxlovid"),
+  # # sotrovimab_complete=covid_therapeutics_complete("Sotrovimab"), 
+  # # molnupiravir_complete =covid_therapeutics_complete("Molnupiravir"),
+  # # paxlovid_complete=covid_therapeutics_complete("Paxlovid"),
+  
+  ## Date treated
+  date_treated = patients.minimum_of(
+    "sotrovimab","paxlovid", "molnupiravir", "remdesivir", "casirivimab"
+  ),
+  drugs_do_not_use = patients.with_these_medications(
+    codelist = drugs_do_not_use_codes, returning = "date", on_or_before = "covid_test_positive_date", find_last_match_in_period = True, date_format = "YYYY-MM-DD",
+  ),
+  drugs_consider_risk = patients.with_these_medications(
+    codelist = drugs_consider_risk_codes, returning = "date",  on_or_before = "covid_test_positive_date", find_last_match_in_period = True, date_format = "YYYY-MM-DD",
+  ),
+
+  # DEMOGRAPHIC COVARIATES & COMORBIDITY
+  age = patients.age_as_of(
+    "covid_test_positive_date", return_expectations = {"rate": "universal", "int": {"distribution": "population_ages"}, "incidence" : 0.9},
+  ),
+  sex = patients.sex(
+    return_expectations = {"rate": "universal", "category": {"ratios": {"M": 0.49, "F": 0.51}},}
+  ),
+  ethnicity = patients.categorised_as(
+            {"Missing": "DEFAULT",
+             "White": "eth='1' OR (NOT eth AND ethnicity_sus='1')", 
+            "Mixed": "eth='2' OR (NOT eth AND ethnicity_sus='2')", 
+            "South Asian": "eth='3' OR (NOT eth AND ethnicity_sus='3')", 
+            "Black": "eth='4' OR (NOT eth AND ethnicity_sus='4')",  
+            "Other": "eth='5' OR (NOT eth AND ethnicity_sus='5')",
+            }, 
+            return_expectations={
+            "category": {"ratios": {"White": 0.6, "Mixed": 0.1, "South Asian": 0.1, "Black": 0.1, "Other": 0.1}},
+            "incidence": 0.4,
+            },
+            ethnicity_sus = patients.with_ethnicity_from_sus(
+                returning="group_6",  
+                use_most_frequent_code=True,
+                return_expectations={
+                    "category": {"ratios": {"1": 0.6, "2": 0.1, "3": 0.1, "4": 0.1, "5": 0.1}},
+                    "incidence": 0.4,
+                    },
+            ),
+            eth=patients.with_these_clinical_events(
+                ethnicity_primis_snomed_codes,
+                returning="category",
+                find_last_match_in_period=True,
+                on_or_before="today",
+                return_expectations={
+                    "category": {"ratios": {"1": 0.6, "2": 0.1, "3": 0.1, "4":0.1,"5": 0.1}},
+                    "incidence": 0.75,
+                },
+            ),
+    ),
+  imd = patients.categorised_as(
```

Comment:
This should be `"Unknown": "DEFAULT",` not 0 -  see https://docs.opensafely.org/study-def-tricks/#grouping-imd-by-quintile

Reply:
changed